pkg/device/nvidia/device.go

```diff
@@ -356,6 +359,67 @@ func (dev *NvidiaGPUDevices) mutateContainerResource(ctr *corev1.Container) bool
 	return false
 }
 
+func (dev *NvidiaGPUDevices) defaultExclusiveCoreIfNeeded(ctr *corev1.Container) bool {
+	if ctr == nil {
+		return false
+	}
+
+	countName := corev1.ResourceName(dev.config.ResourceCountName)
+	if countName == "" || !resourcePresent(ctr, countName) {
+		return false
+	}
+
+	coreName := corev1.ResourceName(dev.config.ResourceCoreName)
+	if coreName == "" || resourcePresent(ctr, coreName) {
+		return false
+	}
+
+	exclusive := false
+	if pct, ok := resourceValue(ctr, corev1.ResourceName(dev.config.ResourceMemoryPercentageName)); ok {
+		exclusive = pct == 100
+	} else if dev.config.ResourceMemoryName == "" {
+		exclusive = true
+	} else if _, ok := resourceValue(ctr, corev1.ResourceName(dev.config.ResourceMemoryName)); !ok {
+		exclusive = true
+	}
+
+	if !exclusive {
+		return false
+	}
+
+	if ctr.Resources.Limits == nil {
+		ctr.Resources.Limits = corev1.ResourceList{}
+	}
+	ctr.Resources.Limits[coreName] = *resource.NewQuantity(100, resource.BinarySI)
```

Comment:
@xrwang8 I think is should be `resource.DecimalSI` rather than `resource.BinarySI` for cores